zerver/lib/test_classes.py

```diff
@@ -1040,12 +1039,14 @@ def api_delete(
 
     def get_streams(self, user_profile: UserProfile) -> List[str]:
         """
-        Helper function to get the stream names for a user
+        Helper function to get the active stream names for a user
         """
-        subs = get_stream_subscriptions_for_user(user_profile).filter(
-            active=True,
+
+        return list(
+            Stream.objects.filter(
+                id__in=get_subscribed_stream_ids_for_user(user_profile)
+            ).values_list("name", flat=True)
         )
-        return [check_string("recipient", get_display_recipient(sub.recipient)) for sub in subs]
```

Comment:
Woah the previous code here was very weird!

Reply:
Haha indeed!